src/components/VersionedCodeEditor/VersionedCodeEditor.tsx

```diff
@@ -0,0 +1,60 @@
+import type { FC } from 'react';
+import { Base64 } from 'js-base64';
+import Loading from 'src/overview/components/Loading';
+
+import { CodeEditor as SDKCodeEditor } from '@openshift-console/dynamic-plugin-sdk';
+import { CodeEditor as PFCodeEditor, Language as PFLanguage } from '@patternfly/react-code-editor';
+import useOpenshiftClusterVersion from '@utils/hooks/useOpenshiftClusterVersion/useOpenshiftClusterVersion';
+
+import { OCP_VERSION_4_18 } from './utils/constants';
+import { isVersionGte } from './utils/utils';
+
+type VersionedCodeEditorProps = {
+  value: string | undefined;
+  onChange: (encodedValue: string) => void;
+  minHeight?: string;
+};
+
+// VersionedCodeEditor component that uses the appropriate CodeEditor based on OCP version
+// https://issues.redhat.com/browse/MTV-2918
+const VersionedCodeEditor: FC<VersionedCodeEditorProps> = ({
+  minHeight = '20rem',
+  onChange,
+  value,
+}) => {
+  const [ocpVersion, loaded] = useOpenshiftClusterVersion();
+
+  const decodedValue = Base64.decode(value ?? '');
+
+  if (!loaded) {
+    return <Loading />;
+  }
+
+  if (isVersionGte(ocpVersion, OCP_VERSION_4_18)) {
```

Comment:
I think it would help to know here why we're checking against v4.18 specifically and when this might need to change.